d3ploy/no_inst.py

```diff
@@ -120,39 +120,53 @@ def __init__(self, *args, **kwargs):
     def enter_notify(self):
         super().enter_notify()
         lib.TIME_SERIES_LISTENERS[self.supply_commod].append(self.extract_supply)
-        lib.TIME_SERIES_LISTENERS[self.demand_commod].append(self.extract_demand)   
-  
+        lib.TIME_SERIES_LISTENERS[self.demand_commod].append(self.extract_demand)
+
     def tick(self):
         """
         This is the tock method for the institution. Here the institution determines the difference
-        in supply and demand and makes the the decision to deploy facilities or not.     
+        in supply and demand and makes the the decision to deploy facilities or not.
         """
         time = self.context.time
         diff, supply, demand = self.calc_diff(time-1)
         if  diff < 0:
             proto = random.choice(self.prototypes)
             ## This is still not correct. If no facilities are present at the start of the
-            ## simulation prod_rate will still return zero. More complex fix is required.            
+            ## simulation prod_rate will still return zero. More complex fix is required.
             if proto in self.fac_supply:
                 prod_rate = self.fac_supply[proto]
             else:
-                print("No facility production rate available for " + proto)                
+                print("No facility production rate available for " + proto)
             number = np.ceil(-1*diff/prod_rate)
             for i in range(int(number)):
                 self.context.schedule_build(self, proto)
                 i += 1
+        if diff > 0:
+            # this is still wonky
+            proto = random.choice(self.prototypes)
+            if proto in self.fac_supply:
+                prod_rate = self.fac_supply[proto]
+            number = np.floor(diff / prod_rate)
+            i = 0
+            while i < number:
+                children = self.children
+                for child in children:
+                    if True:
```

Comment:
Isn't this always true? What's the reasoning on this line?

Reply:
So this is really wonky, but I couldn't figure out a way to access the very first child of the `no_inst` (the facility that has been deployed the earliest) without going through all that. I tried simply indexing `children[0]` but it didn't work. If you know a way, please let me know.

Reply:
True. That is superfluous. I'll get rid of that!